scenarios/options.py

```diff
@@ -0,0 +1,53 @@
+import asyncio
+
+from useragent import Client, Server, Request
+
+
+async def server(ua):
+    invite = await ua.recv_request('INVITE')
+    await invite.respond('100 Trying')
+    await invite.respond('180 Ringing')
+    await invite.respond('200 OK')
+    await ua.recv_request('ACK')
+
+    await ua.send_request('OPTIONS')
+    await ua.recv_response('200 OK')
+
+    await asyncio.sleep(1)
+
+    await ua.send_request('BYE')
+    await ua.recv_response('200 OK')
+
+
+async def client(ua):
+    await ua.send_request('INVITE')
+    response = await ua.recv_response('200 OK', ignore=[100, 180, 183])
+    await response.ack()
+
+    bye = await ua.recv_request('BYE')
+    await bye.respond('200 OK')
+
+
+async def options():
+    async def on_recv(agent, msg):
+        if msg.method == 'OPTIONS':
+            await Request(agent, msg).respond('200 OK')
+            return True
+        return False
+
+    uac = Client(to_uri=f'"sut" <sip:service@127.0.0.1:59361>',
+                 from_uri=f'"sipp" <sip:sipp@127.0.0.1:47398>',
+                 contact_uri=f'sip:service@127.0.0.1:47398',
+                 on_recv=on_recv)
```

Comment:
I was hoping to see something more like a router in here:

```python
uac = Client(....)
uac.router.add_option(handle_option)
```

Instead of providing every single message one by one. We don't need a callback for that, `uac.recv(...)` already works (or should, it might not be there yet).

I want registering callbacks to be easy and straightforward.